File: week1/ferris_wheel.py
n, x = map(int, input().split())
weights = list(map(int, input().split()))
weights.sort()

i, j = 0, n - 1
gondolas = 0

while i <= j:
    if weights[i] + weights[j] <= x:
        i += 1
    j -= 1
    gondolas += 1

print(gondolas)

File: KRISHNA_24b2129/week1/Ferris Wheel.cpp
#include <iostream>
#include <algorithm>
using namespace std;
int main(){
    int n,x=10,gondolas=0;
    cin>>n>>x;
int arr[n];
for(int i=0;i<n;i++){
    cin>>arr[i];
}
sort(arr,arr+n);
int start=0,end=n-1;
while(start<=end){
    if(arr[start]+arr[end]<=x){
        gondolas++;
        start++;
        end--;
    }
    else{
        gondolas++;
        end--;
    }
}
cout<<gondolas;
}